### PR TITLE
First time generate .NET SDK for application insights

### PR DESCRIPTION
**Pull request by frankguodongchen**
## Description
Generate .NET SDK for application insights, it include APIS for manage components, webtests, continuous export configuration and pricing/features.

The related swagger PR is:
https://github.com/Azure/azure-rest-api-specs-pr/pull/139
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [y] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [y] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [y] Title of the pull request is clear and informative.
- [y] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [y] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [y ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [y ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [y ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>ApplicationInsights.Tests</PackageId>
+    <Description>ApplicationInsights.Tests Class library</Description>
+    <PackageId>ApplicationInsights.Tests</PackageId>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+    <TargetFrameworks>net452;</TargetFrameworks>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Remove="BasicTests\**" />
```
Comment (shahabhijeet) [inline src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj]: @frankguodongchen what is the purpose of this ItemGroup?
Please remove as all files/directories are by default included in the project

File: src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    <EmbeddedResource Remove="BasicTests\**" />
+    <None Remove="BasicTests\**" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Update="SessionRecords\**\*.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net452' ">
```
Comment (shahabhijeet) [inline src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj]: This is redundant, please remove.
All these are already included in the project.

File: src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,30 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>ApplicationInsights.Tests</PackageId>
+    <Description>ApplicationInsights.Tests Class library</Description>
+    <PackageId>ApplicationInsights.Tests</PackageId>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+    <TargetFrameworks>net452;</TargetFrameworks>
```
Comment (shahabhijeet) [inline src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj]: Every SDK test should be targeting netCore 1.1, any reason this needs to be target .NET 452?

File: src/SDKs/ApplicationInsights/Management.ApplicationInsights/Microsoft.Azure.Management.ApplicationInsights.csproj [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <PropertyGroup>
+    <Description>Microsoft Azure Application Insights Library</Description>
+    <VersionPrefix>0.1.0-preview</VersionPrefix>
+    <AssemblyName>Microsoft.Azure.Management.ApplicationInsights</AssemblyName>
+
```
Comment (shahabhijeet) [inline src/SDKs/ApplicationInsights/Management.ApplicationInsights/Microsoft.Azure.Management.ApplicationInsights.csproj]: remove extra line

File: src/SDKs/ApplicationInsights/Management.ApplicationInsights/Microsoft.Azure.Management.ApplicationInsights.csproj [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    <PackageId>Microsoft.Azure.Management.ApplicationInsights</PackageId>
+    <PackageTags>Management.ApplicationInsights;</PackageTags>
```
Comment (shahabhijeet) [inline src/SDKs/ApplicationInsights/Management.ApplicationInsights/Microsoft.Azure.Management.ApplicationInsights.csproj]: Add < PackageReleaseNotes > in .csproj
https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/CognitiveServices/dataPlane/Language/Microsoft.CognitiveServices.Language/Microsoft.CognitiveServices.Language.csproj#L11

File: src/SDKs/dirs.proj [hunk 6]
```diff
@@ -47,6 +47,7 @@
 		<ProjectToBuild Include="$(MSBuildThisFileDirectory)StreamAnalytics\Management.StreamAnalytics\Microsoft.Azure.Management.StreamAnalytics.csproj" />
 		<ProjectToBuild Include="$(MSBuildThisFileDirectory)TrafficManager\Management.TrafficManager\Microsoft.Azure.Management.TrafficManager.csproj" />
 		<ProjectToBuild Include="$(MSBuildThisFileDirectory)WebSites\Management.Websites\Microsoft.Azure.Management.Websites.csproj" /> 
+		<ProjectToBuild Include="$(MSBuildThisFileDirectory)ApplicationInsights\Management.ApplicationInsights\Microsoft.Azure.Management.ApplicationInsights.csproj" />
```
Comment (shahabhijeet) [inline src/SDKs/dirs.proj]: This is not needed anymore. This file should not be edited

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Comment by shahabhijeet** [comment]
Travis is failing, that needs to be looked at as to why before this is ready to be merged.

File: src/SDKs/ApplicationInsights/Management.ApplicationInsights/Microsoft.Azure.Management.ApplicationInsights.csproj (hunk 5 above)
Comment (frankguodongchen) [reply src/SDKs/ApplicationInsights/Management.ApplicationInsights/Microsoft.Azure.Management.ApplicationInsights.csproj]: Added, thanks!

File: src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj (hunk 3 above)
Comment (frankguodongchen) [reply src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj]: updated. thanks!

File: src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj (hunk 1 above)
Comment (frankguodongchen) [reply src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj]: OK, updated.

File: src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj (hunk 2 above)
Comment (frankguodongchen) [reply src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj]: removed.

File: src/SDKs/ApplicationInsights/Management.ApplicationInsights/Microsoft.Azure.Management.ApplicationInsights.csproj (hunk 4 above)
Comment (frankguodongchen) [reply src/SDKs/ApplicationInsights/Management.ApplicationInsights/Microsoft.Azure.Management.ApplicationInsights.csproj]: removed. thanks!

File: src/SDKs/dirs.proj (hunk 6 above)
Comment (frankguodongchen) [reply src/SDKs/dirs.proj]: reverted the change. thanks!

File: src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>ApplicationInsights.Tests</PackageId>
+    <Description>ApplicationInsights.Tests Class library</Description>
+    <PackageId>ApplicationInsights.Tests</PackageId>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+    <TargetFrameworks>netcoreapp1.1</TargetFrameworks>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="[1.1.3-preview,2.0.0)" />
```
Comment (shahabhijeet) [inline src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj]: Please use the latest version of Resource Manager nuget package, the latest I believe is 1.6.0

File: src/SDKs/Batch/Management/Management.Batch.Tests/Management.Batch.Tests.csproj [hunk 8]
```diff
@@ -13,6 +13,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="[1.1.3-preview,2.0.0)" />
     <ProjectReference Include="..\Management.Batch\Microsoft.Azure.Management.Batch.csproj" />
+    <PackageReference Update="Microsoft.NET.Test.Sdk" Version="15.3.0" />
```
Comment (shahabhijeet) [inline src/SDKs/Batch/Management/Management.Batch.Tests/Management.Batch.Tests.csproj]: Why are you changing batch.Tests projects?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Batch/Management/Management.Batch.Tests/Management.Batch.Tests.csproj (hunk 8 above)
Comment (frankguodongchen) [reply src/SDKs/Batch/Management/Management.Batch.Tests/Management.Batch.Tests.csproj]: it is accidently changed. updated. thanks!

File: src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj (hunk 7 above)
Comment (frankguodongchen) [reply src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj]: @shahabhijeet it is a range, do we want to make the change. I saw other projects doing the same thing. 
And some project is using <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="1.6.0-preview" />

So should I use 1.6.0-preview or 1.6.0 directly? thanks!

Comment (frankguodongchen) [reply src/SDKs/ApplicationInsights/ApplicationInsights.Tests/ApplicationInsights.Tests.csproj]: @shahabhijeet updated. thanks!

File: src/SDKs/Batch/Management/Management.Batch.Tests/Management.Batch.Tests.csproj [hunk 9]
```diff
@@ -12,7 +12,7 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="[1.1.3-preview,2.0.0)" />
-    <ProjectReference Include="..\Management.Batch\Microsoft.Azure.Management.Batch.csproj" />
+    <ProjectReference Include="..\Management.Batch\Microsoft.Azure.Management.Batch.csproj" />
```
Comment (shahabhijeet) [inline src/SDKs/Batch/Management/Management.Batch.Tests/Management.Batch.Tests.csproj]: Frank I see this file is still being changed.
Can you remove this unwanted file change from your commit?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Batch/Management/Management.Batch.Tests/Management.Batch.Tests.csproj (hunk 9 above)
Comment (frankguodongchen) [reply src/SDKs/Batch/Management/Management.Batch.Tests/Management.Batch.Tests.csproj]: This was a mistakenly touched in my branch and there is no changes in this file.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**